himanshu/postgresAssignment/src/services/studentServices.ts

```diff
@@ -0,0 +1,222 @@
+/* eslint-disable no-tabs */
+/* eslint-disable max-len */
+const sql = require('../sqlserver.ts');
+
+async function addStudent(sid: string, name: string, gender: string, phone: string, classId: string) {
+  return new Promise((resolve, reject) => {
+    sql.query('begin')
+      .then(() => sql.query(
+        'INSERT INTO school.student(SID,NAME,GENDER,PHONE) VALUES ($1,$2,$3,$4)',
+        [sid, name, gender, phone],
+      ))
+      .then(() => sql.query(
+        'INSERT INTO school.studies_in(SID,CLASSID) VALUES ($1,$2)',
+        [sid, classId],
+      ))
+      .then(() => sql.query('commit'))
+      .then(() => {
+        resolve('student added');
+      })
+      .catch((err: any) => {
+        reject(err);
+        return sql.query('rollback');
+      });
+  });
+}
+
+async function addTeacher(tid: string, name: string, gender: string, phone: string, subjectId: string) {
+  return new Promise((resolve, reject) => {
+    sql.query('begin')
+      .then(() => sql.query(
+        'INSERT INTO school.teacher(TID,NAME,GENDER,PHONE) VALUES ($1,$2,$3,$4)',
+        [tid, name, gender, phone],
+      ))
+      .then(() => sql.query(
+        'INSERT INTO school.teacher_takes(TID,SUBJECTID) VALUES ($1,$2)',
+        [tid, subjectId],
+      ))
+      .then(() => sql.query('commit'))
+      .then(() => {
+        resolve('Teacher added');
+      })
+      .catch((err: any) => {
+        reject(err);
+        return sql.query('rollback');
+      });
+  });
+}
+
+async function addSubject(subjectId: string, subjectName: string) {
+  const query = `
+     INSERT INTO school.subject(SUBJECTID,SUBJECTNAME) VALUES($1,$2);
```

Comment:
can you read about ORM's like https://sequelize.org/master/, this give rich functions to handle db operation.